crates/bevy_math/src/cubic_splines.rs

```diff
@@ -546,6 +580,40 @@ impl<P: Point> CubicCurve<P> {
     }
 }
 
+/// A curve that has been divided into straight line segments.
+#[derive(Clone, Debug)]
+pub struct RectifiedCurve {
+    // Each index stores the distance from the start of the curve to that point.
+    arc_lengths: Vec<f32>,
+}
+
+impl RectifiedCurve {
+    /// Total length of the curve.
+    pub fn length(&self) -> f32 {
+        self.arc_lengths[self.arc_lengths.len() - 1]
+    }
+
+    /// Returns a 't' value between 0..=1 based on the distance along the curve.
+    pub fn distance_to_parameter(&self, distance: f32) -> f32 {
```

Comment:
Yeah, definitely don't love the name here. Honestly, it feels like `t(distance)` may actually be the most clear.